src/sympc/protocol/falcon/falcon.py

```diff
@@ -105,14 +107,22 @@ def mul_master(
 
         result = None
 
+        ring_size = int(x.share_ptrs[0].get_ring_size().get_copy())
```

Comment:
Q: Why not take the `ring_size` from `MPCTensor`, `session` attribute?

Reply:
We operate in three ring_size (2,67,session_rings(2**64,2**32)), an MPC Tensor might be either of three, the ring_size is needed here for beaver triples to generate binary shares, and in truncation and several other places . We are moving taking all the config info at the tensor level instead of session level.This approach reduced several re-encoding issues and makes it compatible with various ring_sizes.

Reply:
Q: But the ```get_ring_size``` where it is in the ```RST``` Tensor? (I do not see it here)

Reply:
https://github.com/OpenMined/SyMPC/pull/269/files#diff-41d43b95b2abaf26d7dffd507ee7a5755e1fcd044e55e0b0485669e8ff9fc622R146